File: src/main/java/com/projet/hopital/entities/Medecin.java
package com.projet.hopital.entities;

import com.projet.hopital.enumeration.Specialite;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data @AllArgsConstructor @NoArgsConstructor
public class Medecin {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nom;
    private String prenom;
    private String email;
    @Enumerated(EnumType.STRING)
    private Specialite specialite;
    private String numeroDeSalle;
    private String telephone;
    private int experienceYears;
    private String qualifications;
    private boolean suppression = false;
}

File: src/web/src/app/model/medecin.ts
export class Medecin {
    id!: number;
    nom!: string;
    prenom!:string;
    email!: string;
    specialite!: Specialite;
    numeroDeSalle!: string;
    telephone!: string;
    experienceYears!: number;
    qualifications!: string;
    suppression: boolean = false;
  }

  export enum Specialite {
    MÉDECINEINTERNE,
    PÉDIATRIE,
    CARDIOLOGIE,
    DERMATOLOGIE,
    CHIRURGIEORTHOPÉDIQUE,
    GASTROENTÉROLOGIE,
    NEUROLOGIE,
    PSYCHIATRIE,
    OPHTALMOLOGIE,
    RHUMATOLOGIE,
    NÉPHROLOGIE
}
  